cli/src/commands/git/fetch.rs

```diff
@@ -35,30 +35,43 @@ use crate::git_util::print_git_import_stats;
 use crate::git_util::with_remote_git_callbacks;
 use crate::ui::Ui;
 
+/// Specifies which branches to fetch from remotes
+#[derive(Clone, Debug)]
+enum BranchesToFetch<'a> {
+    /// Fetch specific branch patterns
+    Patterns(&'a [StringPattern]),
+    /// Fetch only tracked branches, optionally filtered by explicit patterns
+    Tracked {
+        patterns: &'a [StringPattern],
+        has_explicit_patterns: bool,
```

Comment:
More minor: Do you really need `has_explicit_patterns`? I think this only affects one warning message, but that warning makes sense either way (or you could compare the patterns to `glob:*`, but I don't think you have to).